src/main/java/util/RefreshTimer.java

```diff
@@ -0,0 +1,101 @@
+package util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * Facilitates the countdown for periodic refreshes.
+ */
+public class RefreshTimer extends TickingTimer{
```

Comment:
Yes, the ApiQuotaManager can keep the timer object.
I don't have a good idea about the event handling flow of the current code, but based on normal design practices, the code for updating the GUI should be in the GUI itself. Do what you think is best for now. We can revisit the event handling again during the summer.